crates/typst-pdf/src/color.rs

```diff
@@ -377,26 +377,37 @@ impl PaintEncode for Color {
 /// Extra color space functions.
 pub(super) trait ColorSpaceExt {
     /// Returns the range of the color space.
-    fn range(self) -> [f32; 6];
+    fn range(self) -> [f32; 8];
 
     /// Converts a color to the color space.
-    fn convert<U: QuantizedColor>(self, color: Color) -> [U; 3];
+    fn convert<U: QuantizedColor>(self, color: Color) -> Vec<U>;
 }
 
 impl ColorSpaceExt for ColorSpace {
-    fn range(self) -> [f32; 6] {
-        [0.0, 1.0, 0.0, 1.0, 0.0, 1.0]
+    fn range(self) -> [f32; 8] {
+        [0.0, 1.0, 0.0, 1.0, 0.0, 1.0, 0.0, 1.0] // Not sure about that
     }
 
-    fn convert<U: QuantizedColor>(self, color: Color) -> [U; 3] {
+    fn convert<U: QuantizedColor>(self, color: Color) -> Vec<U> {
```

Comment:
Using a `Vec` is undesirable here since it's used in many places, it should be a fixed sized array otherwise every single color being encoded will cause an allocation. Perhaps a `SmallVec<[U; 3]>` could be okay.

Reply:
Agreed, again. Not sure what the best option is as, if I' not mistaken, the number of elements depends on the color space. Or would it make sense to return a `[U: 4]` in all cases?·

Reply:
Tentative solution returning a `SmallVec<[U; 4]>`:

https://github.com/typst/typst/pull/4568/files#diff-55650bf2dd7b33ec02c37fd3b314eb3a5db0c2aa6bf0baf269ef9c03fe716c95L391-R413

Again, not about performance (I have very little experience with `SmallVec`s); please let me know if you think a different array size or return type would be better.